internal/gojsonschema/utils.go

```diff
@@ -131,41 +100,39 @@ const (
 )
 
 func mustBeInteger(what interface{}) *int {
+	number, ok := what.(json.Number)
+	if !ok {
+		return nil
+	}
 
-	if isJSONNumber(what) {
-
-		number := what.(json.Number)
-
-		isInt := checkJSONInteger(number)
-
-		if isInt {
-
-			int64Value, err := number.Int64()
-			if err != nil {
-				return nil
-			}
-
-			int32Value := int(int64Value)
-			return &int32Value
-		}
+	isInt := checkJSONInteger(number)
+	if !isInt {
+		return nil
+	}
 
+	int64Value, err := number.Int64()
+	if err != nil {
+		return nil
 	}
 
-	return nil
+	// This doesn't actually convert to an int32 value; it converts to the
```

Comment:
As noted in the comment, the code suggests that it converts the value to an int32, but it is incorrect. On 64-bit systems this is kept as a 64-bit integer, but in 32-bit systems it is converted to a 32-bit integer. I'm unsure what we actually want to do here.